packages/canary-web-components/src/components/ic-menu-with-multi/ic-menu.tsx

```diff
@@ -0,0 +1,1276 @@
+          }
+      }
+    }
+  };
+
+  private setInputValue = (highlightedOptionIndex: number) => {
+    const menuOptions = this.getMenuOptions();
+
+    if (menuOptions[highlightedOptionIndex] !== undefined) {
+      this.menuOptionSelect.emit({
+        value: menuOptions[highlightedOptionIndex][this.valueField],
+      });
+
+      if (this.closeOnSelect) {
+        this.optionHighlighted = undefined;
+        this.menuOptionId.emit({ optionId: undefined });
+      }
+    }
+
+    if (this.closeOnSelect) {
+      if (!this.hasTimedOut) {
+        this.handleMenuChange(false);
+      } else {
+        (this.parentEl as HTMLIcSearchBarElement).setFocus();
+      }
+    }
+  };
+
+  private handleOptionClick = (event: Event): void => {
+    const { value, label } = (event.target as HTMLLIElement).dataset;
+    this.menuOptionSelect.emit({ value, label });
+    this.optionHighlighted = undefined;
+
+    if (this.isMultiSelect) {
+      this.multiOptionClicked = value;
+    } else {
+      this.handleMenuChange(false);
+    }
+  };
+
+  private handleRetry = (): void => {
+    this.retryButtonClicked.emit({ value: this.value });
+  };
+
+  private handleRetryKeyDown = (ev: KeyboardEvent): void => {
+    if (ev.key === "Enter" || ev.key === " ") {
+      ev.preventDefault();
+      this.retryButtonClicked.emit({ value: this.value, keyPressed: ev.key });
+    }
+  };
+
+  private handleBlur = (event: FocusEvent): void => {
+    if (event.relatedTarget !== this.inputEl) {
+      if (event.relatedTarget === this.selectAllButton) {
+        this.menu.removeAttribute(this.activeDescendantAttr);
+      }
+
+      if (
+        !(
+          this.menu.contains(event.relatedTarget as HTMLElement) ||
+          event.relatedTarget === this.selectAllButton
+        )
+      ) {
+        this.handleMenuChange(false, this.hasPreviouslyBlurred);
+        this.menu.removeAttribute(this.activeDescendantAttr);
+      }
+    } else {
+      this.handleMenuChange(false);
+      this.preventClickOpen = true;
+      this.menu.removeAttribute(this.activeDescendantAttr);
+    }
+    if (!this.isSearchBar) this.hasPreviouslyBlurred = !!event.relatedTarget;
+  };
+
+  private handleMouseDown = (event: Event): void => {
+    event.preventDefault();
+  };
+
+  private handleMenuKeyDown = (event: KeyboardEvent) => {
+    if (this.activationType === "automatic") {
+      this.autoSetValueOnMenuKeyDown(event);
+    } else if (this.activationType === "manual" && !this.isSearchBar) {
+      this.manualSetInputValueKeyboardOpen(event);
+    }
+  };
+
+  private handleMenuKeyUp = (event: KeyboardEvent): void => {
+    if (event.key === "Tab" && event.shiftKey) {
+      this.preventClickOpen = false;
+    }
+    if (event.key === "Enter" && this.disabledOptionSelected) {
+      this.disabledOptionSelected = false;
+      event.stopImmediatePropagation();
+    }
+  };
+
+  private handleSelectAllClick = () => {
+    this.keyboardNav = false;
+    this.menu.focus();
```

Comment:
Naturally, if I had 3,4 or 5 out of 15 options selected and I wanted to clear them all, I would go to the select all button and double press to clear it. I wouldn't expect the focus to go back on the menu. By going back to the menu, it suggests I need to individually select/deselect when I could use the button again. 

This could be a personal preference for usability of the select all/clear all button

Reply:
That is a good point. I have walked through this keyboard behaviour with a designer and they are happy with it, however. Like you say, it depends on personal preference and is subjective - it personally seems ok to me, especially as the user can just tab back onto the button again pretty quickly.

I wonder if we leave it as it is for now and see if we get any feedback about this once it has been released as a canary component?